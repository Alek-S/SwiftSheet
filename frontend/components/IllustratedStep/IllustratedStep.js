import React from 'react';
import styled from 'styled-components';

/** @function
 * @name IllustratedStep
 * Step with header body text and provided illustration;
 *
 * @returns {JSX}
 */
const IllustratedStep = ({ header, graphicPath, body, step, children }) => {
	return (
		<StyledSection>
			<img src={graphicPath} alt="illustration" width="180" />
			<header>
				<h2>
					<span className="step">{step}</span> {header}
				</h2>
			</header>
			<p>{children}</p>
		</StyledSection>
	);
};

const StyledSection = styled.section`
	margin: auto;
	padding-top: 4rem;
	padding-bottom: 4rem;

	img {
		float: left;
		margin-right: 4rem;
	}
	header {
		padding-top: 10px;
	}

	h2 {
		font-weight: 500;
		font-size: 1.4rem;
		letter-spacing: 1px;
		line-height: 175%;
	}

	p {
		margin-top: 1.5rem;
		margin-left: 20.5rem;
		font-size: 1rem;
		line-height: 175%;
		max-width: 600px;
	}

	a {
		color: ${props => props.theme.color.red};
	}

	.step {
		border: solid 1px ${props => props.theme.color.text};
		border-radius: 40px;
		padding: 0.5rem 1.5rem;
		font-weight: 400;
		margin-right: 0.75rem;
	}

	.highlight {
		/* background-color: rgba(255, 255, 0, .2); */
		color: #333;
		font-weight: 500;
		/* padding-left: .3rem;
		padding-right: .3rem; */
	}

	@media (max-width: 1200px) {
		width: 90%;
	}
	@media (max-width: 900px) {
		width: 98%;

		p {
			margin-top: 0.5rem;
			margin-left: 12rem;
			line-height: 150%;
		}
	}
`;

export { IllustratedStep };
