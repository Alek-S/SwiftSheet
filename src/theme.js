/**
 * default themes for use with styled-components
 * passed in with theme provider
 */
const theme = {
	boxShadow: `0px 5px 40px -5px rgba(0,0,0,0.3)`,
	boxShadowDark: `0px 5px 40px -2px rgba(0,0,0,0.5)`,
	color: {
		background: '#fafafa',
		backgroundDark: '#EAEAEA',
		backgroundDarkest: '#BCBCBC',
		border: '#d9d9d9',
		text: '#404040',
		lightText: '#848484',
		red: '#ED5050',
		lightRed: '#FF8D8D',
		green: '#D8E13F',
		blue: '#46C8F5',
		lightBlue: '#97E4FF',
		darkBlue: '#2E8AA8',
	},
	gradient: {
		light: `linear-gradient(to bottom, #ffffff 0%, #f6f6f6 47%, #ededed 100%)`,
		red: 'linear-gradient(to bottom, #f07878 0%, #ED5050 100%)',
		lightRed: 'linear-gradient(to bottom, #f39191 0%, #ef6262 100%)',
		blue: 'linear-gradient(to bottom, #D8E13F 0%, #BFC644 100%)',
		greenBlue: ' linear-gradient(160deg, #57DACB 0%, #55A2ED 100%)',
	},
	font: {
		header: `'Poiret One', cursive`,
		main: `'Quicksand', sans-serif`,
	},
};

export default theme;
