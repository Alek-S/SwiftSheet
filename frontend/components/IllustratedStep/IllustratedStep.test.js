import React from 'react';
import 'jest-styled-components';

import { IllustratedStep } from './IllustratedStep';
import theme from '../../theme';

describe('Footer', () => {
	test('matches the snapshot', () => {
		const tree = shallow(<IllustratedStep theme={theme} />);
		expect(toJson(tree)).toMatchSnapshot();
	});
});
