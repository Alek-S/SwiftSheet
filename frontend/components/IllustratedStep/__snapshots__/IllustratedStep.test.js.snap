// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer matches the snapshot 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "IllustratedStep__StyledSection-sc-108b0jo-0",
        "isStatic": false,
        "rules": Array [
          "margin:auto;padding-top:4rem;padding-bottom:4rem;img{float:left;margin-right:4rem;}header{padding-top:10px;}h2{font-weight:500;font-size:1.4rem;letter-spacing:1px;line-height:175%;}p{margin-top:1.5rem;margin-left:20.5rem;font-size:1rem;line-height:175%;max-width:600px;}a{color:",
          [Function],
          ";}.step{border:solid 1px ",
          [Function],
          ";border-radius:40px;padding:0.5rem 1.5rem;font-weight:400;margin-right:0.75rem;}.highlight{color:#333;font-weight:600;}@media (max-width:1200px){width:90%;}@media (max-width:900px){width:98%;p{margin-top:0.5rem;margin-left:12rem;line-height:150%;}}@media (max-width:750px){img{display:block;float:none;width:40%;margin:auto;}p{margin-left:0.5rem;}}",
        ],
      },
      "displayName": "IllustratedStep__StyledSection",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "IllustratedStep__StyledSection-sc-108b0jo-0",
      "target": "section",
      "toString": [Function],
      "usesTheme": true,
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
>
  <img
    alt="illustration"
    width="180"
  />
  <header>
    <h2>
      <span
        className="step"
      />
       
    </h2>
  </header>
  <p />
</StyledComponent>
`;
